diff --git a/config/common.mk b/config/common.mk
index 73ee74e..62f9cf3 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -180,16 +180,11 @@ PRODUCT_PACKAGES += \
 
 # Custom CM packages
 PRODUCT_PACKAGES += \
-    ResurrectionOTA \
-    ResurrectionStats \
-    Trebuchet \
     AudioFX \
-    CMFileManager \
     Eleven \
     LockClock \
     CMSettingsProvider \
     ExactCalculator \
-    Jelly \
     LiveLockScreenService \
     WeatherProvider \
     OmniStyle \
@@ -200,13 +195,10 @@ PRODUCT_PACKAGES += \
 
 WITH_ROOT_METHOD ?= rootless
 ifeq ($(WITH_ROOT_METHOD), magisk)
-# Magisk Manager
-PRODUCT_PACKAGES += \
-    MagiskManager
-
-# Copy Magisk zip
+# SuperSU
 PRODUCT_COPY_FILES += \
-    vendor/cm/prebuilt/common/magisk.zip:system/addon.d/magisk.zip
+    vendor/cm/prebuilt/common/UPDATE-SuperSU.zip:system/addon.d/UPDATE-SuperSU.zip \
+    vendor/cm/prebuilt/common/etc/init.d/99SuperSUDaemon:system/etc/init.d/99SuperSUDaemon
 endif
 
 # Exchange support
@@ -312,9 +304,9 @@ DEVICE_PACKAGE_OVERLAYS += vendor/cm/overlay/common
 
 PRODUCT_VERSION = 5.8.5
 ifneq ($(RR_BUILDTYPE),)
-RR_VERSION := RR-N-v$(PRODUCT_VERSION)-$(shell date -u +%Y%m%d)-$(CM_BUILD)-$(RR_BUILDTYPE)
+RR_VERSION := RR-N-v$(PRODUCT_VERSION)-$(shell date +%Y%m%d)-$(CM_BUILD)-$(RR_BUILDTYPE)
 else
-RR_VERSION := RR-N-v$(PRODUCT_VERSION)-$(shell date -u +%Y%m%d)-$(CM_BUILD)
+RR_VERSION := RR-N-v$(PRODUCT_VERSION)-$(shell date +%Y%m%d)-$(CM_BUILD)
 endif
 
 PRODUCT_PROPERTY_OVERRIDES += \
diff --git a/config/common.mk b/config/common.mk
index a9e9c3e..531a749 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -91,6 +91,18 @@ PRODUCT_COPY_FILES += \
 PRODUCT_COPY_FILES += \
     vendor/cm/prebuilt/DeskClock/DeskClock.apk:system/app/DeskClock/DeskClock.apk
 
+# Screenshot App
+PRODUCT_COPY_FILES += \
+    vendor/cm/prebuilt/common/app/Screenshot.apk:system/app/Screenshot/Screenshot.apk
+
+# YahooWeatherProvider App
+PRODUCT_COPY_FILES += \
+    vendor/cm/prebuilt/common/app/YahooWeatherProvider.apk:system/app/YahooWeatherProvider/YahooWeatherProvider.apk
+
+# KernelAdiutor App
+PRODUCT_COPY_FILES += \
+    vendor/cm/prebuilt/common/app/KernelAdiutor.apk:system/priv-app/KernelAdiutor/KernelAdiutor.apk
+
 # DU Utils Library
 PRODUCT_BOOT_JARS += \
     org.dirtyunicorns.utils
diff --git a/gello/Android.mk b/gello/Android.mk
new file mode 100644
index 0000000..e078577
--- /dev/null
+++ b/gello/Android.mk
@@ -0,0 +1,51 @@
+#
+# Copyright (C) 2016 The CyanogenMod Project
+#           (C) 2017 The LineageOS Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH := $(call my-dir)
+
+SOURCE_GELLO_PATH := $(LOCAL_PATH)/../../../external/gello-build
+
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := Gello
+LOCAL_MODULE_CLASS := APPS
+LOCAL_CERTIFICATE := $(DEFAULT_SYSTEM_DEV_CERTIFICATE)
+
+LOCAL_OVERRIDES_PACKAGES := Jelly
+
+ifeq ($(WITH_GELLO_SOURCE),true)
+# Build from source
+ifeq ($(LOCAL_GELLO),true)
+BUILD_GELLO := $(info $(shell bash $(SOURCE_GELLO_PATH)/gello_build.sh --local 1>&2))
+else
+BUILD_GELLO := $(info $(shell bash $(SOURCE_GELLO_PATH)/gello_build.sh 1>&2))
+endif
+LOCAL_SRC_FILES := ../../../external/gello-build/Gello.apk
+include $(BUILD_PREBUILT)
+else
+
+LOCAL_DEX_PREOPT := false
+LOCAL_MODULE_TAGS := optional
+LOCAL_BUILT_MODULE_STEM := package.apk
+LOCAL_MODULE_SUFFIX := $(COMMON_ANDROID_PACKAGE_SUFFIX)
+
+LOCAL_HTTP_FILE_VERSION := 40
+LOCAL_HTTP_PATH := https://github.com/LineageOS/android_packages_apps_Gello/releases/download/$(LOCAL_HTTP_FILE_VERSION)
+LOCAL_HTTP_FILENAME := gello.apk
+LOCAL_HTTP_MD5SUM := $(LOCAL_HTTP_FILENAME).md5sum
+
+include $(BUILD_HTTP_PREBUILT)
+endif
diff --git a/gello/Android.mk b/gello/Android.mk
index e078577..a88c42a 100644
--- a/gello/Android.mk
+++ b/gello/Android.mk
@@ -24,7 +24,7 @@ LOCAL_MODULE := Gello
 LOCAL_MODULE_CLASS := APPS
 LOCAL_CERTIFICATE := $(DEFAULT_SYSTEM_DEV_CERTIFICATE)
 
-LOCAL_OVERRIDES_PACKAGES := Jelly
+LOCAL_OVERRIDES_PACKAGES := Browser
 
 ifeq ($(WITH_GELLO_SOURCE),true)
 # Build from source
