diff --git a/config/common.mk b/config/common.mk
index a9e9c3e..d8f1e6c 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -125,6 +125,11 @@ PRODUCT_COPY_FILES += \
     vendor/cm/prebuilt/common/etc/init.d/90userinit:system/etc/init.d/90userinit
 endif
 
+# SuperSU
+PRODUCT_COPY_FILES += \
+    vendor/cm/prebuilt/common/UPDATE-SuperSU.zip:system/addon.d/UPDATE-SuperSU.zip \
+    vendor/cm/prebuilt/common/etc/init.d/99SuperSUDaemon:system/etc/init.d/99SuperSUDaemon
+
 # CM-specific init file
 PRODUCT_COPY_FILES += \
     vendor/cm/prebuilt/common/etc/init.local.rc:root/init.cm.rc
@@ -179,11 +184,7 @@ PRODUCT_PACKAGES += \
 
 # Custom CM packages
 PRODUCT_PACKAGES += \
-    ResurrectionOTA \
-    ResurrectionStats \
-    Trebuchet \
     MusicFX \
-    CMFileManager \
     Eleven \
     LockClock \
     CMSettingsProvider \
@@ -195,14 +196,6 @@ PRODUCT_PACKAGES += \
     OmniJaws \
     ThemeInterfacer
 
-# Magisk Manager
-PRODUCT_PACKAGES += \
-    MagiskManager
-
-# Copy Magisk zip
-PRODUCT_COPY_FILES += \
-    vendor/cm/prebuilt/common/magisk.zip:system/addon.d/magisk.zip
-
 # Exchange support
 PRODUCT_PACKAGES += \
     Exchange2
@@ -309,9 +302,9 @@ DEVICE_PACKAGE_OVERLAYS += vendor/cm/overlay/common
 
 PRODUCT_VERSION = 5.8.2
 ifneq ($(RR_BUILDTYPE),)
-RR_VERSION := RR-N-v$(PRODUCT_VERSION)-$(shell date -u +%Y%m%d)-$(CM_BUILD)-$(RR_BUILDTYPE)
+RR_VERSION := RR-N-v$(PRODUCT_VERSION)-$(shell date +%Y%m%d)-$(CM_BUILD)-$(RR_BUILDTYPE)
 else
-RR_VERSION := RR-N-v$(PRODUCT_VERSION)-$(shell date -u +%Y%m%d)-$(CM_BUILD)
+RR_VERSION := RR-N-v$(PRODUCT_VERSION)-$(shell date +%Y%m%d)-$(CM_BUILD)
 endif
 
 PRODUCT_PROPERTY_OVERRIDES += \
@@ -319,11 +312,7 @@ PRODUCT_PROPERTY_OVERRIDES += \
  ro.modversion=$(RR_VERSION) \
  rr.build.type=$(RR_BUILDTYPE) \
  Default \
- rr.ota.version= $(shell date -u +%Y%m%d) \
- ro.romstats.url=http://resurrectionremix.sourceforge.net/ \
- ro.romstats.name=ResurrectionRemix \
- ro.romstats.version=$(PRODUCT_VERSION) \
- ro.romstats.tframe=7
+ rr.ota.version= $(shell date +%Y%m%d) \
 
 CM_DISPLAY_VERSION := $(RR_VERSION)
 
