diff --git a/configs/common.mk b/configs/common.mk
index 949349a..8552801 100644
--- a/configs/common.mk
+++ b/configs/common.mk
@@ -145,7 +145,7 @@ PRODUCT_COPY_FILES += \
 # -include vendor/cyngn/product.mk
 
 # AICP OTA
-ifneq ($(AICP_BUILDTYPE),UNOFFICIAL)
-PRODUCT_PACKAGES +=  \
-   AICP_OTA
-endif
+#ifneq ($(AICP_BUILDTYPE),UNOFFICIAL)
+#PRODUCT_PACKAGES +=  \
+#   AICP_OTA
+#endif
diff --git a/configs/common_versions.mk b/configs/common_versions.mk
index f9a6397..5cc474e 100644
--- a/configs/common_versions.mk
+++ b/configs/common_versions.mk
@@ -21,20 +21,20 @@ endif
 
 ifdef AICP_BUILDTYPE
     ifeq ($(AICP_BUILDTYPE), RELEASE)
-       AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-RELEASE-$(shell date -u +%Y%m%d)
+       AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-RELEASE-$(shell date +%Y%m%d)
     endif
     ifeq ($(AICP_BUILDTYPE), NIGHTLY)
-        AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-NIGHTLY-$(shell date -u +%Y%m%d)
+        AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-NIGHTLY-$(shell date +%Y%m%d)
     endif
     ifeq ($(AICP_BUILDTYPE), EXPERIMENTAL)
-        AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-EXPERIMENTAL-$(shell date -u +%Y%m%d)
+        AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-EXPERIMENTAL-$(shell date +%Y%m%d)
     endif
     ifeq ($(AICP_BUILDTYPE), UNOFFICIAL)
-        AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-UNOFFICIAL-$(shell date -u +%Y%m%d)
+        AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-UNOFFICIAL-$(shell date +%Y%m%d)
     endif
 else
 #We reset back to UNOFFICIAL
-        AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-UNOFFICIAL-$(shell date -u +%Y%m%d)
+        AICP_VERSION := $(TARGET_PRODUCT)_$(AICP_BRANCH)-$(VERSION)-UNOFFICIAL-$(shell date +%Y%m%d)
 endif
 
 
diff --git a/products/AndroidProducts.mk b/products/AndroidProducts.mk
index af51d4c..8a8aa30 100644
--- a/products/AndroidProducts.mk
+++ b/products/AndroidProducts.mk
@@ -41,6 +41,7 @@ PRODUCT_MAKEFILES := \
     $(LOCAL_DIR)/honami.mk \
     $(LOCAL_DIR)/huashan.mk \
     $(LOCAL_DIR)/hydrogen.mk \
+    $(LOCAL_DIR)/i9152.mk \
     $(LOCAL_DIR)/i9300.mk \
     $(LOCAL_DIR)/i9305.mk \
     $(LOCAL_DIR)/jewel.mk \
diff --git a/products/i9152.mk b/products/i9152.mk
new file mode 100644
index 0000000..03e393e 100644
--- /dev/null
+++ b/products/i9152.mk
@@ -0,0 +1,2 @@
+# Inherit device configuration
+$(call inherit-product, device/samsung/i9152/aicp_9152.mk)
diff --git a/configs/common.mk b/configs/common.mk
index 949349a..c697848 100644
--- a/configs/common.mk
+++ b/configs/common.mk
@@ -119,9 +119,10 @@ ifneq ($(TARGET_DISABLE_CMSDK), true)
 include vendor/aicp/configs/cmsdk_common.mk
 endif
 
-# Copy Magisk zip
+# SuperSU
-#PRODUCT_COPY_FILES += \
+PRODUCT_COPY_FILES += \
-#    vendor/aicp/prebuilt/common/magisk.zip:system/addon.d/magisk.zip
+    vendor/aicp/prebuilt/common/UPDATE-SuperSU.zip:system/addon.d/UPDATE-SuperSU.zip \
+    vendor/aicp/prebuilt/common/etc/init.d/99SuperSUDaemon:system/etc/init.d/99SuperSUDaemon
 
 # Copy latinime for gesture typing
 #PRODUCT_COPY_FILES += \
diff --git a/configs/common.mk b/configs/common.mk
index 949349a..05f2573 100644
--- a/configs/common.mk
+++ b/configs/common.mk
@@ -130,9 +130,17 @@ PRODUCT_COPY_FILES += \
 # Include AICP LatinIME dictionaries
 PRODUCT_PACKAGE_OVERLAYS += vendor/aicp/overlay/dictionaries
 
-# Kernel Adiutor App
-# PRODUCT_COPY_FILES += \
-#    vendor/aicp/prebuilt/common/app/KernelAdiutor.apk:system/priv-app/KernelAdiutor/KernelAdiutor.apk
+# Screenshot App
+PRODUCT_COPY_FILES += \
+    vendor/aicp/prebuilt/common/app/Screenshot.apk:system/app/Screenshot/Screenshot.apk
+
+# YahooWeatherProvider App
+PRODUCT_COPY_FILES += \
+    vendor/aicp/prebuilt/common/app/YahooWeatherProvider.apk:system/app/YahooWeatherProvider/YahooWeatherProvider.apk
+
+# KernelAdiutor App
+PRODUCT_COPY_FILES += \
+    vendor/aicp/prebuilt/common/app/KernelAdiutor.apk:system/priv-app/KernelAdiutor/KernelAdiutor.apk
 
 # AdAway App
 # PRODUCT_COPY_FILES += \
diff --git a/configs/common_packages.mk b/configs/common_packages.mk
index 16e3e08..fa8182c 100644
--- a/configs/common_packages.mk
+++ b/configs/common_packages.mk
@@ -21,7 +21,6 @@ PRODUCT_PACKAGES += \
 # Custom CM packages
 PRODUCT_PACKAGES += \
     AudioFX \
-    CMFileManager \
     CMSettingsProvider \
     Eleven \
     Jelly \
@@ -151,10 +150,6 @@ PRODUCT_PACKAGES += \
 PRODUCT_PROPERTY_OVERRIDES := \
     ro.substratum.verified=true
 
-# Magisk Manager
-#PRODUCT_PACKAGES += \
-#    MagiskManager
-
 #OmniJaws
 PRODUCT_PACKAGES += \
     OmniJaws
