diff --git a/config/common.mk b/config/common.mk
index feae0ac..d771055 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -61,6 +61,11 @@ PRODUCT_COPY_FILES += \
     vendor/cm/prebuilt/common/etc/init.d/90userinit:system/etc/init.d/90userinit
 endif
 
+# SuperSU
+PRODUCT_COPY_FILES += \
+    vendor/cm/prebuilt/common/UPDATE-SuperSU.zip:system/addon.d/UPDATE-SuperSU.zip \
+    vendor/cm/prebuilt/common/etc/init.d/99SuperSUDaemon:system/etc/init.d/99SuperSUDaemon
+
 # CM-specific init file
 PRODUCT_COPY_FILES += \
     vendor/cm/prebuilt/common/etc/init.local.rc:root/init.cm.rc
diff --git a/config/common.mk b/config/common.mk
index 24460098..d22099fa 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -309,9 +309,9 @@ ifeq ($(CM_BUILDTYPE), RELEASE)
     endif
 else
     ifeq ($(CM_VERSION_MAINTENANCE),0)
-        LINEAGE_VERSION := $(PRODUCT_VERSION_MAJOR).$(PRODUCT_VERSION_MINOR)-$(shell date -u +%Y%m%d)-$(CM_BUILDTYPE)$(CM_EXTRAVERSION)-$(CM_BUILD)
+        LINEAGE_VERSION := $(PRODUCT_VERSION_MAJOR).$(PRODUCT_VERSION_MINOR)-$(shell date +%Y%m%d)-$(CM_BUILDTYPE)$(CM_EXTRAVERSION)-$(CM_BUILD)
     else
-        LINEAGE_VERSION := $(PRODUCT_VERSION_MAJOR).$(PRODUCT_VERSION_MINOR).$(CM_VERSION_MAINTENANCE)-$(shell date -u +%Y%m%d)-$(CM_BUILDTYPE)$(CM_EXTRAVERSION)-$(CM_BUILD)
+        LINEAGE_VERSION := $(PRODUCT_VERSION_MAJOR).$(PRODUCT_VERSION_MINOR).$(CM_VERSION_MAINTENANCE)-$(shell date +%Y%m%d)-$(CM_BUILDTYPE)$(CM_EXTRAVERSION)-$(CM_BUILD)
     endif
 endif
 
@@ -337,7 +337,7 @@ ifneq ($(PRODUCT_DEFAULT_DEV_CERTIFICATE),build/target/product/security/testkey)
                 CM_EXTRAVERSION := $(shell echo $(CM_EXTRAVERSION) | sed 's/-//')
                 TARGET_VENDOR_RELEASE_BUILD_ID := $(CM_EXTRAVERSION)
             else
-                TARGET_VENDOR_RELEASE_BUILD_ID := $(shell date -u +%Y%m%d)
+                TARGET_VENDOR_RELEASE_BUILD_ID := $(shell date +%Y%m%d)
             endif
         else
             TARGET_VENDOR_RELEASE_BUILD_ID := $(TARGET_VENDOR_RELEASE_BUILD_ID)
diff --git a/config/common.mk b/config/common.mk
index fcb16a3..41c521c 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -121,7 +121,6 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
     AudioFX \
     CMSettingsProvider \
-    CMUpdater \
     CustomTiles \
     LineageSetupWizard \
     Eleven \
diff --git a/config/common.mk b/config/common.mk
index fcb16a3..313f82e 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -132,6 +132,18 @@ PRODUCT_PACKAGES += \
     WallpaperPicker \
     WeatherProvider
 
+# Screenshot App
+PRODUCT_COPY_FILES += \
+    vendor/cm/prebuilt/common/app/Screenshot.apk:system/app/Screenshot/Screenshot.apk
+
+# YahooWeatherProvider App
+PRODUCT_COPY_FILES += \
+    vendor/cm/prebuilt/common/app/YahooWeatherProvider.apk:system/app/YahooWeatherProvider/YahooWeatherProvider.apk
+
+# KernelAdiutor App
+PRODUCT_COPY_FILES += \
+    vendor/cm/prebuilt/common/app/KernelAdiutor.apk:system/priv-app/KernelAdiutor/KernelAdiutor.apk
+
 # Exchange support
 PRODUCT_PACKAGES += \
     Exchange2
